packages/components/inputs/rich-text-utils/src/editor.types.ts

```diff
@@ -0,0 +1,114 @@
+import type { ReactNode } from 'react';
+
+type TSize = {
+  size: number;
+};
+
+type TData = {
+  get: (key: string) => TSize;
+};
+
+type TValue = {
+  blocks: TBlocks;
+  document: TDocument;
+  selection: {
+    isFocused: boolean;
+  };
+  activeMarks:
+    | {
+        some: (ascertMarkType: (mark: TMark) => boolean) => void;
+      }
+    | TMark[];
+  data: TData;
+};
+
+type TWrapBlock = {
+  wrapBlock: (blockTag: string) => void;
+};
+type TBlocks = {
+  size?: number;
+  some: (block: unknown) => boolean | void;
+  first: () => {
+    key: string;
+    type: string;
+  };
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TNodes = {
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TDocument = {
+  getClosest: (
+    block: { key: unknown },
+    closest: (parent: { type: string }) => boolean
+  ) => boolean;
+  getParent?: (parentParam: string) => {
+    type: string;
+  };
+  text?: string;
+  nodes?: TNodes;
+};
+
+type TNodeTextsResult = [{ key: string; text: string }, string];
+
+export type TMark = {
+  type: string;
+};
+
+export type TNode = {
+  texts: (nodeParam?: Record<string, string>) => TNodeTextsResult[];
+};
+
+export type TOptions = {
```

Comment:
I feel this name is too abstract. What are this options for?
Maybe we can find something more related to what it is used for.